lib/modulr/api/notifications_service.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Modulr
+  module API
+    class NotificationsService < Service
+      def find(id:, **opts)
+        _response = client.get("#{base_notification_url(opts)}/notifications/#{id}")
+      end
+
+      def list(**opts)
+        _response = client.get("#{base_notification_url(opts)}/notifications")
+      end
+
+      def create(type:, channel:, destinations:, config:, **opts)
+        payload = {
+          type: type,
+          channel: channel,
+          destinations: destinations,
+          config: config,
+        }
+        _response = client.post("#{base_notification_url(opts)}/notifications", payload)
+      end
+
+      protected def base_notification_url(opts)
+        opts[:partner_id] ? "/partners/#{opts[:partner_id]}" : "/customers/#{opts[:customer_id]}"
```

Comment:
Notifications can be registered at two different levels: the partner level and the customer level. The API is the same for both, so I've created only one service for both:

```rb
client.notifications.create(partner_id: "P001", type: "foo", etc...)
# and
client.notifications.create(customer_id: "C001", type: "foo", etc...)
```

I don't expect that behaviour to change so I think dealing with both endpoints in the same service is the correct tradeoff.